test/unit/codegen/codegen_llvm_ir.cpp

```diff
@@ -367,12 +367,12 @@ SCENARIO("Function", "[visitor][llvm]") {
             std::smatch m;
 
             // Check function signature. The return type should be the default double type.
-            std::regex function_signature(R"(define double @foo\(double %x1\) \{)");
+            std::regex function_signature(R"(define double @foo\(double %x[0-9].*\) \{)");
```

Comment:
Before merging this, I have a question for @georgemitenkov for this change.
I got issues on my system (Ubuntu 20.04, LLVM 11) when I run the tests because the variables had different names. Instead of `x1` for example it was `x6`, etc. Do you agree with this change in the tests?
I am not familiar with how LLVM decides which number to append to the variable name, so I was wondering if this is a reasonable change to avoid test failing from system to system.

Reply:
LLVM auto-index the variables if there is a collision in the symbol table. The change is fine (maybe `%x[0-9]+`?)! the reason I had `x1` was to match the assignment of the temporary with its usage:
```
%t = ...
... = ... %t, ...
```
I do not see a way of checking this unless we "hardcode" the temporary SSA names (which is pretty bad in my opinion). I guess testing manually and using regexes like `[0-9]+` should be ok. In future, we may want to integrate FileCheck (https://llvm.org/docs/CommandGuide/FileCheck.html) - they have a nice way of matching and storing a temporary for usage checks.

Reply:
I see. Thanks a lot for the clarification once more! :+1: